src/lib/resource/sections/rep-confirmation-method.js

```diff
@@ -0,0 +1,247 @@
+            ],
+            definitions: {
+                'if-email-then-q-rep-enter-your-email-address-is-required': {
+                    if: {
+                        properties: {
+                            'q-rep-confirmation-method': {
+                                const: 'email'
+                            }
+                        },
+                        required: ['q-rep-confirmation-method']
+                    },
+                    then: {
+                        required: ['q-rep-enter-your-email-address'],
+                        propertyNames: {
+                            enum: ['q-rep-confirmation-method', 'q-rep-enter-your-email-address']
+                        },
+                        errorMessage: {
+                            required: {
+                                'q-rep-enter-your-email-address': 'Enter an email address'
+                            }
+                        }
+                    }
+                },
+                'if-text-then-q-rep-enter-your-telephone-number-is-required': {
+                    if: {
+                        properties: {
+                            'q-rep-confirmation-method': {
+                                const: 'text'
+                            }
+                        },
+                        required: ['q-rep-confirmation-method']
+                    },
+                    then: {
+                        required: ['q-rep-enter-your-telephone-number'],
+                        propertyNames: {
+                            enum: ['q-rep-confirmation-method', 'q-rep-enter-your-telephone-number']
+                        },
+                        errorMessage: {
+                            required: {
+                                'q-rep-enter-your-telephone-number':
+                                    'Enter a UK mobile phone number'
+                            }
+                        }
+                    }
+                },
+                'if-none-then-phone-and-email-explicitly-not-required': {
+                    if: {
+                        properties: {
+                            'q-rep-confirmation-method': {
+                                const: 'none'
+                            }
+                        },
+                        required: ['q-rep-confirmation-method']
+                    },
+                    then: {
+                        additionalProperties: false,
+                        properties: {
+                            'q-rep-confirmation-method': {
+                                const: 'none'
+                            }
+                        },
+                        required: ['q-rep-confirmation-method']
+                    }
+                }
+            },
+            errorMessage: {
+                required: {
+                    'q-rep-confirmation-method':
+                        'Select how you want to get your confirmation message'
```

Comment:
@armoj should this error mirror the question content? ie 'Select how we should tell you we've got your application'

Reply:
fixed